### ISSUE TITLE
Favorites and Additional Setting for Safe Mode

### ISSUE THREAD

**Issue by maisans-maid** [nullxception/boorusphere#9]
Awesome app btw. Loved it! This is objectively the best open-source booru app that i've found on github (after hours of searching, installing, and uninstalling app). I'd like to ask if the following feature could possibly be implemented within the app.

1. **Favorite/Bookmark** - Allow us to bookmark images on the booru so that we have the option to get back to these images later without downloading it. I have no clue how fetching on android app works and if the API of these websites allow for fetching multiple images at once (assuming this works by saving the url in the device), tho i assume this could be done by adding the option to link booru accounts (and use their favorite feature)

and...

2. **Strict/Loose Options for Safe Mode** - Strict option would not fetch the image (the current safe mode of the app). Loose option would fetch but blur the image preview (kinda like how reddit blurs the nsfw) and opening it would prompt a warning. Not essential but i think it would be awesome to implement it in the app.

**Comment by nullxception** [comment]
I think it's fairly easy to implement on flutter side, however we have to implement the API key usage (#4) first before we can fetch favorite data from booru sites.
Pretty nice idea, added to my plan, thank you!

**Comment by maisans-maid** [comment]
Hey, if its ok can I bother you with an additional feature? since a new version hasn't been released yet, can you make it so that the downloads section have the option to display the list as images. We can't treally distinguish the files from the filename alone since most of the time its random alphanumeric characters like the one below:

![Screenshot_2022-07-24-16-36-21-732_io.chaldeaprjkt.boorusphere.jpg](https://user-images.githubusercontent.com/56829176/180639378-51a34922-3dad-411b-81ce-ddf0c2bc744f.jpg)

Also, if it can be done, it'd be better if you can group them according to the server they are downloaded from 😁

- Thanks for your hard work 💪👍

**Comment by nullxception** [comment]
> can you make it so that the downloads section have the option to display the list as images. 
> Also, if it can be done, it'd be better if you can group them according to the server they are downloaded from 😁

another nice idea! I'll try to implement it later, thanks!

**Comment by maisans-maid** [comment]
Is it by design that questionable rating are not blurred? If so, can we have the option to blur questionable ratings as well. Something like adding a radio-button style on which rating to blur (with explicit always turned on)

**Comment by nullxception** [comment]
> Is it by design that questionable rating are not blurred? If so, can we have the option to blur questionable ratings as well. Something like adding a radio-button style on which rating to blur (with explicit always turned on)

yes, booru sites uses questionable rating as default, it means the post is waiting to be rated properly as safe or explicit.
thus I'm not going to add another option for that.

**Comment by maisans-maid** [comment]
I see, never knew that was the rating they used for unrated media. Thanks for the heads up!

**Comment by nullxception** [comment]
> I see, never knew that was the rating they used for unrated media. Thanks for the heads up!
> 
> EDIT: Is adding another option to safe mode that allows you to just not fetch unrated content (questionable) viable? Like yes you can fetch explicit content and blur them but not fetch questionable ones at all.

yes, in fact, that "strict safe mode" is only fetch content rated as safe.
but well, again... except safebooru.

**Comment by nullxception** [comment]
Favorites feature is postponed for now since I have no plan for it in short time, so I'll mark this issues as postponed and close it,
